tests/unit-core/test-api.c

```diff
@@ -145,7 +145,7 @@ handler_construct (const jerry_value_t func_obj_val, /**< function object */
   TEST_ASSERT (jerry_get_boolean_value (args_p[0]) == true);
 
   jerry_value_t field_name = jerry_create_string ((jerry_char_t *) "value_field");
-  jerry_set_property (this_val, field_name, args_p[0]);
+  jerry_release_value (jerry_set_property (this_val, field_name, args_p[0]));
```

Comment:
This might hide erros if we release the value without any check in the unit tests. Please use `TEST_ASSERT` to verify the return value of the API funtion is a boolean value and not an error.  Fix similar cases below.